includes/wizards/class-network-wizard.php

```diff
@@ -55,93 +61,203 @@ public function __construct() {
+			// Display header.
+			$this->admin_header_init(
+				[
+					'title' => $this->get_name(),
+					'tabs'  => $this->get_tabs(),
+				]
+			);
+		}
 	}
 
 	/**
-	 * Current screen callback to detect Network admin screens.
+	 * Get the name for this current screen's wizard. Required by parent abstract.
+	 *
+	 * @return string The wizard name.
 	 */
-	public function current_screen() {
+	public function get_name() {
+		return esc_html( $this->admin_screens[ $this->get_screen_slug() ] );
+	}
 
-		global $current_screen, $plugin_page;
+	/**
+	 * Get slug if we're currently viewing a Network screen.
+	 * 
+	 * @return string
+	 */
+	private function get_screen_slug() {
+		
+		global $pagenow;
 
-		// Check for admin page screen type.
-		if ( isset( $this->admin_screens['page'][ $plugin_page ] ) ) {
-			$this->slug = $plugin_page;
-			$this->screen_type = 'page';
-			return $this->enable_display();
+		static $screen_slug;
+		if ( isset( $screen_slug ) ) {
+			return $screen_slug;
 		}
+		$screen_slug = '';
 
-		// Check for admin post type screen: Listings page and classic editor (add new + edit), but not block editor.
-		if ( ! empty( $current_screen->post_type )
-			&& ! empty( $this->admin_screens['post_type'][ $current_screen->post_type ] )
-			&& false === $current_screen->is_block_editor ) {
-			$this->slug = $current_screen->post_type;
-			$this->screen_type = 'post_type';
-			return $this->enable_display();
+		$sanitized_action    = sanitize_text_field( $_GET['action'] ?? '' ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		$sanitized_page      = sanitize_text_field( $_GET['page'] ?? '' ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		$sanitized_post_type = sanitize_text_field( $_GET['post_type'] ?? '' ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		$sanitized_post_id   = sanitize_text_field( $_GET['post'] ?? '' ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		
+		if ( 'admin.php' === $pagenow && isset( $this->admin_screens[ $sanitized_page ] ) ) {
+			// admin page screen: admin.php?page={page} .
+			$screen_slug = $sanitized_page;
+		} elseif ( 'edit.php' === $pagenow && isset( $this->admin_screens[ $sanitized_post_type ] ) ) {
+			// post type list screen: edit.php?post_type={post_type} .
+			$screen_slug = $sanitized_post_type;
+		} elseif ( 'post-new.php' === $pagenow && 'newspack_hub_nodes' === $sanitized_post_type ) {
+			// add new node screen: post-new.php?post_type=newspack_hub_nodes .
+			// note: assumes non-block editor, otherwise we need to not set this.
+			$screen_slug = $sanitized_post_type;
+		} elseif ( 'post.php' === $pagenow && 'edit' === $sanitized_action && 'newspack_hub_nodes' === get_post_type( $sanitized_post_id ) ) {
+			// edit node screen: post.php?post={ID}&action=edit
+			// note: assumes non-block editor, otherwise we need to not set this.
+			$screen_slug = 'newspack_hub_nodes';
 		}
+
```

Comment:
Readability improvement/recommendation. Below uses PHP8 `match` expression, switch can also be used.
```suggestion
		$screen_slug = match ( true ) {
			'admin.php' === $pagenow && isset( $this->admin_screens[ $sanitized_page ] ) 
				=> $sanitized_page,
			'edit.php' === $pagenow && isset( $this->admin_screens[ $sanitized_post_type ] ) 
				=> $sanitized_post_type,
			'post-new.php' === $pagenow && 'newspack_hub_nodes' === $sanitized_post_type 
				=> $sanitized_post_type,
			'post.php' === $pagenow && 'edit' === $sanitized_action && 'newspack_hub_nodes' === get_post_type( $sanitized_post_id ) 
				=> 'newspack_hub_nodes',
			default => '',
		};
```

Reply:
Thanks!  Updated and resolved.